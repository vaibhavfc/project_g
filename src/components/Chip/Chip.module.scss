@import '../../sass/theme/default';

//MEDIUM SIZE STYLES
.rootMedium {
  height: 1.125rem;
  line-height: 0.625rem;
  padding: 0.25rem 0.5rem;
  border-radius: 0.25rem;
  color: $chip-text__default;
  background-color: $chip-bg__default;
  @include subtitle;

  & > * {
    padding: 0;
  }
}

.successMedium {
  background-color: $chip-bg--medium__success;
  color: $white;
}

.pendingMedium {
  background-color: $chip-bg--medium__pending;
  color: $white;
}

.warningMedium {
  background-color: $chip-bg--medium__warning;
  color: $white;
}

.info {
  background-color: $chip-bg__info;
  color: $chip-text__info;
}

//SMALL SIZE STYLES
.rootSmall {
  height: 1.125rem;
  line-height: 0.875rem;
  padding: 0.25rem 0.5rem;
  border-radius: 0.25rem;
  color: $chip-text__default;
  background-color: $neutral-blue;
  @include info-1;
  text-transform: capitalize;

  & > * {
    padding: 0;
  }
}

.successSmall {
  background-color: $chip-bg--small__success;
  color: $chip-bg--medium__success;
}

.pendingSmall {
  background-color: $chip-bg--small__success;
  color: $chip-bg--medium__pending;
}

.warningSmall {
  background-color: $chip-bg--small__warning;
  color: $chip-bg--medium__warning;
}

.info {
  background-color: $chip-bg__info;
  color: $chip-text__info;
}
