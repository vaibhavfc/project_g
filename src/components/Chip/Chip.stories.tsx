// also exported from '@storybook/react' if you can deal with breaking changes in 6.1
import { Story, Meta } from '@storybook/react/types-6-0';

import Chip from '.';

import classes from './LinkWithClass.module.scss'

import { ChipProps } from './Chip.type';

export default {
  title: 'Components/Chip',
  component: Chip,
  parameters: {
    controls: {
      // disabled: true,
    }
  }
} as Meta;

const Template: Story<ChipProps> = (args) => <Chip {...args} />

export const DefaultMediumChip = Template.bind({});
DefaultMediumChip.args = {
  label: 'Default',
  size: 'medium',
  type: 'default'
}

export const DefaultSmallChip = Template.bind({});
DefaultSmallChip.args = {
  label: 'Default',
  size: 'small',
  type: 'default'
}

export const SuccessMediumChip = Template.bind({});
SuccessMediumChip.args = {
  label: 'Success',
  size: 'medium',
  type: 'success'
}

export const SuccessSmallChip = Template.bind({});
SuccessSmallChip.args = {
  label: 'Success',
  size: 'small',
  type: 'success'
}

export const PendingMediumChip = Template.bind({});
PendingMediumChip.args = {
  label: 'Pending',
  size: 'medium',
  type: 'pending'
}

export const PendingSmallChip = Template.bind({});
PendingSmallChip.args = {
  label: 'Pending',
  size: 'small',
  type: 'pending'
}

export const WarningMediumChip = Template.bind({});
WarningMediumChip.args = {
  label: 'Warning',
  size: 'medium',
  type: 'warning'
}

export const WarningSmallChip = Template.bind({});
WarningSmallChip.args = {
  label: 'Warning',
  size: 'small',
  type: 'warning'
}
export const InfoMediumChip = Template.bind({});
InfoMediumChip.args = {
  label: 'Info',
  size: 'medium',
  type: 'info'
}

export const InfoSmallChip = Template.bind({});
InfoSmallChip.args = {
  label: 'Info',
  size: 'small',
  type: 'info'
}

