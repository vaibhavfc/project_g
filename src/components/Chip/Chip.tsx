import React, { FC, forwardRef, useMemo } from 'react';
import { Chip as MuiChip } from '@material-ui/core';
import classnames from 'classnames';
import { ChipProps, ChipStylingProps } from './Chip.type';

import classes from './Chip.module.scss';

import { mergeClassesObjects } from '../../helpers/styling/mergeClassesObjects';

const ChipWithInnerRef: FC<ChipProps> = ({
  // innerRef = null,
  classes: overrideClasses = {},
  // children = 'Chip',
  variant = 'default',
  size,
  label,
  type,
  // doneCallback = () => undefined,
  // ...props
}) => {
  const mergedClasses = useMemo(
    () => mergeClassesObjects<ChipStylingProps>(classes, overrideClasses),
    [overrideClasses],
  );
  return (
    <MuiChip
      label={label}
      variant={variant}
      size={size}
      classes={{
        root: classnames(
          mergedClasses.rootMedium, {
            [classes.successMedium]: (size === 'medium' && type === 'success'),
            [classes.pendingMedium]: (size === 'medium' && type === 'pending'),
            [classes.warningMedium]: (size === 'medium' && type === 'warning'),
            [classes.info]: (size === 'medium' && type === 'info'),
          },
        ),
        sizeSmall: classnames(
          mergedClasses.rootSmall, {
            [classes.successSmall]: (size === 'small' && type === 'success'),
            [classes.pendingSmall]: (size === 'small' && type === 'pending'),
            [classes.warningSmall]: (size === 'small' && type === 'warning'),
            [classes.info]: (size === 'small' && type === 'info'),
          },
        ),
      }}
    />
  );
};

const Chip = forwardRef<HTMLDivElement, Omit<ChipProps, 'innerRef'>>(
  (props, ref) => <ChipWithInnerRef innerRef={ref} {...props} />,
);
export default Chip;
