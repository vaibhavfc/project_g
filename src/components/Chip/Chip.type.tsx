import { Ref } from 'react'
import { ChipProps as MuiChipProps, ChipClassKey } from '@material-ui/core/Chip'
export interface ChipStylingProps extends Partial<Record<ChipClassKey, string>> {
  rootMedium?: string
  rootSmall?: string
}

export interface ChipProps extends MuiChipProps {
  innerRef: Ref<HTMLDivElement>;
  type: 'default' | 'success' | 'pending' | 'warning' | 'info';
  doneCallback?: () => void;
}
