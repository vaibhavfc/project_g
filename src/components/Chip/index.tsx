import Chip from './Chip';
import { ChipProps } from './Chip.type';

export type { ChipProps }

export default Chip;