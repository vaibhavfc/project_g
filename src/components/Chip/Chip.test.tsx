import { render, screen, cleanup } from '@testing-library/react';
import React from 'react';

import Chip from './Chip';

beforeEach(cleanup);

describe('Chip', () => {
  it('Renders Default Nudge', () => {
    render(<Chip type="default" chipType="status" label="Label" />);
    expect(screen.getByTestId('chip-status-default').classList.contains('MuiChip-root')).toBe(true);
  });

  it('Renders Default Chip', () => {
    render(<Chip type="default" chipType="nudge" label="Label" />);
    expect(screen.getByTestId('chip-nudge-default').classList.contains('MuiChip-sizeSmall')).toBe(true);
  });

  it('Success Status', () => {
    render(<Chip type="success" chipType="status" label="Label" />);
    expect(screen.getByTestId('chip-status-success').classList.contains('MuiChip-root')).toBe(true);
  });

  it('Pending Nudge', () => {
    render(<Chip type="pending" chipType="nudge" label="Label" />);
    expect(screen.getByTestId('chip-nudge-pending').classList.contains('MuiChip-sizeSmall')).toBe(true);
  });
  it('Warning Status', () => {
    render(<Chip type="warning" chipType="status" label="Label" />);
    expect(screen.getByTestId('chip-status-warning').classList.contains('MuiChip-root')).toBe(true);
  });

  it('Info Nudge', () => {
    render(<Chip type="info" chipType="nudge" label="Label" />);
    expect(screen.getByTestId('chip-nudge-info').classList.contains('MuiChip-sizeSmall')).toBe(true);
  });

  it('Renders correct laber', () => {
    render(<Chip label="Label" />);
    expect(screen.getByText('Label')).toBeVisible();
  });
});
