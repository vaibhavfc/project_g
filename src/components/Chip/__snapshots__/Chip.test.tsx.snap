// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Matches Snapshot of Chip 1`] = `
<div
  className="MuiChip-root"
  onKeyDown={[Function]}
  onKeyUp={[Function]}
>
  <span
    className="MuiChip-label"
  />
</div>
`;
